src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1PlacementApplicationDomainEventService.kt

```diff
@@ -181,4 +185,51 @@ class Cas1PlacementApplicationDomainEventService(
       ),
     )
   }
+
+  fun placementApplicationAssessed(
+    placementApplication: PlacementApplicationEntity,
+    assessedByUser: UserEntity,
+    placementApplicationDecision: PlacementApplicationDecisionEnvelope,
+  ) {
+    val domainEventId = UUID.randomUUID()
+    val eventOccurredAt = Instant.now()
+    val application = placementApplication.application
+
+    val assessor = assessedByUser.let { domainEventTransformer.toStaffMember(it) }
+
+    val assessmentDecision = when (placementApplication.decision) {
+      PlacementApplicationDecision.ACCEPTED -> RequestForPlacementAssessed.Decision.accepted
+      PlacementApplicationDecision.REJECTED -> RequestForPlacementAssessed.Decision.rejected
+      null -> throw IllegalArgumentException("PlacementApplicationDecision was null")
+      else -> throw IllegalArgumentException("PlacementApplicationDecision '${placementApplication.decision}' has been deprecated")
```

Comment:
I think it may be a tiny bit more resilient to future changes if we explicitly include the two remaining cases instead of using an 'else'. Outcome for both would be identical of course. You can specify both cases on one line using a comma to separate them to reduce code duplication, see https://kotlinlang.org/docs/control-flow.html#when-expression

Reply:
Good call.   I had them separately but ran into a code complexity issue.